pkg/internal/resources/apiserver/apiserver.go

```diff
@@ -48,142 +51,220 @@ var k = kustomize.NewDefaultKustomize()
-								},
-								{
-									"name":  "API_SERVER_OIDC_CLIENT_ID",
-									"value": c.Spec.OIDC.ClientID,
-								},
-								{
-									"name":  "API_SERVER_OIDC_CA_FILE",
-									"value": "/run/oidc-certs/ca.crt",
-								},
+								"--authentication-mode=$(AUTHENTICATION_MODE)",
+							},
+							Env: []corev1.EnvVar{
 								{
-									"name":  "API_SERVER_OIDC_USERNAME_CLAIM",
-									"value": c.Spec.OIDC.UsernameClaim,
+									Name:  "API_SERVER_ADDR",
+									Value: ":8443",
 								},
 								{
-									"name":  "API_SERVER_OIDC_USERNAME_PREFIX",
-									"value": c.Spec.OIDC.UsernamePrefix,
+									Name:  "API_SERVER_TLS_CERT_FILE",
+									Value: "/run/serving-certs/tls.crt",
 								},
 								{
-									"name":  "API_SERVER_OIDC_GROUPS_CLAIM",
-									"value": c.Spec.OIDC.GroupsClaim,
+									Name:  "API_SERVER_TLS_PRIVATE_KEY_FILE",
+									Value: "/run/serving-certs/tls.key",
 								},
 								{
-									"name":  "API_SERVER_OIDC_GROUPS_PREFIX",
-									"value": c.Spec.OIDC.GroupsPrefix,
+									Name:  AuthModeEnv,
+									Value: "",
 								},
+							},
+							VolumeMounts: []corev1.VolumeMount{
 								{
-									"name":  "API_SERVER_OIDC_SIGNING_ALGS",
-									"value": strings.Join(c.Spec.OIDC.SupportedSigningAlgs, ","),
+									MountPath: "/run/serving-certs",
+									ReadOnly:  true,
+									Name:      "serving-cert",
 								},
 							},
-							"volumeMounts": []map[string]interface{}{{
-								"mountPath": "/run/serving-certs",
-								"readyOnly": true,
-								"name":      "serving-cert",
-							}, {
-								"mountPath": "/run/oidc-certs",
-								"readyOnly": true,
-								"name":      "oidc-cert",
-							}},
 						},
 					},
-					"volumes": []map[string]interface{}{{
-						"name": "serving-cert",
-						"secret": map[string]interface{}{
-							"secretName": c.Spec.TLSSecretRef.Name,
-						},
-					}, {
-						"name": "oidc-cert",
-						"secret": map[string]interface{}{
-							"secretName": c.Spec.OIDC.CertificateAuthority.Name,
+					Volumes: []corev1.Volume{
+						{
+							Name: "serving-cert",
+							VolumeSource: corev1.VolumeSource{
+								Secret: &corev1.SecretVolumeSource{
+									SecretName: c.Spec.TLSSecretRef.Name,
+								},
+							},
 						},
-					}},
+					},
 				},
 			},
 		},
 	}
-	managerEnvBytes, err := yaml.Marshal(managerEnv)
+	if c.Spec.StaticUsers != nil {
+		containers := deploymentPatch.Spec.Template.Spec.Containers
+		for i, container := range containers {
+			if container.Name == "manager" {
+				containers[i].Args = append(containers[i].Args,
+					"--htpasswd-secret-name=$(HTPASSWD_SECRET_NAME)",
+				)
+				containers[i].Env = append(containers[i].Env,
+					corev1.EnvVar{
+						Name:  "HTPASSWD_SECRET_NAME",
+						Value: c.Spec.StaticUsers.HtpasswdSecret.Name,
+					},
+				)
+				supportedAuth = append(supportedAuth, "Htpasswd")
+				for j, env := range containers[i].Env {
```

Comment:
why not just add this `AuthModeEnv` at the end, once all valid authorization modes have been processed?

Reply:
fixed